src/components/counter/counter.tsx

```diff
@@ -1,64 +1,62 @@
-import { Popconfirm, Progress, Slider } from "antd";
 import * as React from "react";
+import { CounterControls } from "./counterControls";
+import { CounterDisplay } from "./counterDisplay";
 interface ICounterState {
-  count: number;
-  timeout: number;
-  run: boolean;
+    count: number;
+    timeout: number;
+    run: boolean;
 }
 
 const counterGoal = 50000;
 
 export class Counter extends React.Component<{}, ICounterState> {
-  public state = { count: 0, timeout: 300, run: true };
-  private interval: any;
+    public state = { count: 0, timeout: 300, run: true };
+    private interval: any;
 
-  constructor(props) {
-    super(props);
-  }
-
-  public componentWillMount() {
-    if (!this.interval) {
-      this.incrementCounter();
+    constructor(props) {
+        super(props);
     }
-  }
 
-  public incrementCounter = () => {
-    this.setState({ count: this.state.count + 1 });
-    if (this.state.run) {
-      this.interval = setTimeout(this.incrementCounter, this.state.timeout);
+    public componentWillMount() {
+        if (!this.interval) {
+            this.incrementCounter();
+        }
     }
-  }
 
-  public onRateChange = (value: number) => {
-    this.setState({ timeout: (100 - value) * 10 });
-  }
+    public incrementCounter = () => {
+        this.setState({ count: this.state.count + 1 });
+        if (this.state.run) {
+            this.interval = setTimeout(this.incrementCounter, this.state.timeout);
+        }
+    }
 
-  public componentWillUnmount() {
-    clearInterval(this.interval);
-  }
+    public onRateChange = (value: number) => {
+        this.setState({ timeout: (100 - value) * 10 });
+    }
 
-  public formatProgress(percent) {
-    return String(Math.floor(percent * (counterGoal / 100)));
-  }
+    public componentWillUnmount() {
+        clearInterval(this.interval);
+    }
 
-  public render() {
-    return (
-      <div>
-        <Progress
-          type="circle"
-          format={this.formatProgress}
-          percent={this.state.count / (counterGoal / 100)}
-        />
-        <Progress type="line" showInfo={false} percent={(this.state.count % 100)} />
-        <h3>Behold the mighty counter!</h3>
-        <p>You are no match for his powers...</p>
-        <p>Your updates cannot destory him...</p>
-        <p>Admire his will to survive...</p>
-        <p>You find yourself helplessly falling in love...</p>
+    public formatProgress(percent) {
+        return String(Math.floor(percent * (counterGoal / 100)));
+    }
 
-        <Slider defaultValue={100 - (this.state.timeout / 10)} onChange={this.onRateChange} />
-      </div>
-    );
-  }
+    public render() {
+        return (
+            <div>
+                <CounterDisplay
+                    circleFormat={this.formatProgress}
+                    circlePercent={this.state.count / (counterGoal / 100)}
+                    linePercent={(this.state.count % 100)}
+                >
+                    <CounterControls
```

Comment:
Seems wierd passing this into as a child of DisplayCounter (I would have just had them as siblings), but good demonstration of `props.children`.